cloudinary-core/src/main/java/com/cloudinary/metadata/AbstractMetadataField.java

```diff
@@ -0,0 +1,113 @@
+package com.cloudinary.metadata;
+
+import org.cloudinary.json.JSONObject;
+
+import java.text.ParseException;
+
+/**
+ * Represents a single metadata field. Use one of the derived classes in the metadata API calls.
+ * @param <T>
+ */
+public class AbstractMetadataField<T> extends JSONObject {
```

Comment:
What's T stands for? the type of the field? 
If so, why do we need TYPE?

Reply:
Kept on reading and I got what you meant. 
but... will that hold for future types that might be added to the metadata?

Reply:
You can already use it for new types (without updating the SDK). I renamed class (removed the abstract) and added a string constructor to make it clearer/easier.